gnosis_vpn-root/src/main.rs

```diff
@@ -286,28 +271,40 @@ async fn loop_daemon(
                         match wg_cmd {
                             WireGuardCommand::WgUp( wg_data ) => {
                                 // ensure we run down before going up to ensure clean slate
-                                if let Some(routing) = routing {
-                                    match routing.teardown().await {
+                                if let Some(router) = maybe_router.take() {
+                                    match router.teardown().await {
                                         Ok(_) => tracing::warn!("cleaned up routing from previous instance"),
-                                        Err(err) => {
-                                            tracing::debug!(error = ?err, "expected error during pre-start routing teardown");
+                                        Err(error) => {
+                                            tracing::debug!(?error, "expected error during pre-start routing teardown");
                                         }
                                     }
                                 }
 
-                                let new_routing = Routing::new(worker_user.clone(), wg_data);
-                                let res = new_routing.setup().await.map_err(|e| format!("routing setup error: {}", e));
-                                *routing = Some(new_routing);
-                                send_to_worker(&IncomingWorker::WgUpResult { res }, &mut writer).await?;
+                                match routing::build_router(worker_user.clone(), wg_data) {
```

Comment:
why make this a Result?

It should be well defined for the supported systems and never fail on construction